src/memcached_cache.cpp

```diff
@@ -615,13 +615,43 @@ Store::Status MemcachedCache::update_irs_associated_impus(MemcachedImplicitRegis
   return status;
 }
 
+// Create an IMPU from an IRS and set it in the store.
+// This is used when we think that the IMPU is new.
+// The behaviour depends on the registration state of the IMPU we're trying to
+// store:
+//  - if we're storing an IMPU in state REGISTERED, then we just blind write to
+//    the store. This is because we only do this when we've got a new
+//    registration, so we've just communicated with the HSS and the data we have
+//    is authoritative. We therefore want to overwrite anything that's been put
+//    in the store since then
```

Comment:
Missing full stop.

Reply:
Also I'd say:

> This is is safe because either:
> 
> a) the data in the store matches what we have, or is equivalently valid, in which case, what we have can just be updated
> b) the data in the store is from an UNREGISTERED request at the same time. Given that we are about to authoritatively tell Sprout we are registered, we should present a consistent registered state in the store. @
>
> Note, we aren't correcting any differing additional IMPUs, or IMPIs. These should have be subject to a PPR, and the HSS, and will expire anyway. The additional latency in resolving this very narrow race (Multiple SAA and HSS change happening simultaneously to a single IMPU) isn't worthwhile resolving, and it's not obvious which one should win.
```